api/src/main/java/org/apache/iceberg/catalog/Namespace.java

```diff
@@ -48,4 +52,10 @@ public static Namespace namespace(String[] levels) {
   public static Namespace empty() {
     return EMPTY;
   }
+
+  @Override
+  public String toString() {
+    // assumes a level it self won't have a period in it, otherwise it will be ambiguous
+    return Arrays.stream(levels).collect(joining("."));
```

Comment:
Is it necessary to use the streams API here? What about `Joiners.on(".").join(levels)`? Seems simpler.